packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -132,7 +132,10 @@ function UncontrolledInnerBlocks( props ) {
 		/>
 	);
 
-	if ( Object.keys( blockType.providesContext ).length === 0 ) {
+	if (
+		! blockType?.providesContext ||
+		Object.keys( blockType.providesContext ).length === 0
```

Comment:
I'm still not fully understanding how it's possible to have inner blocks without a block type. Could you give an example?

Reply:
Honestly, I am not sure how this is happening either! But, it happens in the scenario mentioned in the description with WooCommerce's use of blocks in the new product editor. Ultimately, the root problem may lie somewhere in WooCommerce's handling of blocks, but this change seems like a reasonable defensive change in GB.

Reply:
Yes, I'm still curious. This is a legitimate error tbh, but sure, let's gracefully fail.

Reply:
Agreed. We will still be attempting to address the root cause of this! I'm very curious too.